src/dmd/backend/cgobj.d

```diff
@@ -1202,67 +1195,38 @@ version (MARS)
                 lastfilename = filename;
             }
         }
+        cseg = ln.cseg;
+        assert(cseg > 0);
+        obj.pubnamidx = ln.seg;
 
-    SameFileLoop:
-        while (1)
-        {
-if (++count >= 10000)
-{
-fprintf(stderr, "infinite loop file %s %d %d %d\n",
-        lastfilename ? lastfilename : "null",
-        cast(int)obj.linnum_list.length,
-        cast(int)li, cast(int)ll);
-assert(0);
-}
-            cseg = ln.cseg;
-            assert(cseg > 0);
-            obj.pubnamidx = ln.seg;
-
-            Srcpos srcpos;
-            version (MARS)
-                srcpos.Sfilename = ln.filename;
-            else
-                srcpos.Sfilptr = &ln.filptr;
-
-            for (uint u = 0; u < ln.i; )
-            {
-                srcpos.Slinnum = *cast(ushort *)&ln.data[u];
-                u += 2;
-                targ_size_t offset;
-                if (I32)
-                    offset = *cast(uint *)&ln.data[u];
-                else
-                    offset = *cast(ushort *)&ln.data[u];
-                OmfObj_linnum(srcpos,cseg,offset);
-                u += _tysize[TYint];
-            }
-            ln.i = 0;
-            linnum_flush();
-            if (ll == li)       // if ll is at the end of the array
-                --li;           // shrink the end of the array
+        Srcpos srcpos;
+        version (MARS)
+            srcpos.Sfilename = ln.filename;
+        else
+            srcpos.Sfilptr = &ln.filptr;
```

Comment:
In a future PR, could this be abstracted at the aggregate level, instead of having all caller code having this `version` ?

Reply:
Sure, but there's a lot like that.